src/supervisor.rs

```diff
@@ -31,7 +35,7 @@ use crate::{host, Tag};
 /// let count2 = hello.request(Count);
 /// assert_eq!(count1, count2);
 /// ```
-pub trait Supervisor
+pub trait Supervisor<S = Bincode>
```

Comment:
I think we also need here `M` or it will only work with a subset of serializers (serde based).

Reply:
I'm not quite sure what `M` generic would be for here, since I thought that's pretty much what `Self::Arg` was?

`Self::Arg` has `: Serialize + DeserializeOwned` before this PR, doesn't that make things still tied to serde?

Reply:
Yes, it's tied to serde right now because we use Bincode. But if we want to support any serializer it can't be tied to it anymore. Or it would be impossible to make it work with Protobuf for example. That's why we can't use `Self::Arg` anymore to express the type, we need a more complicated `where S: Serializer<M>` telling the compiler that `M` can be any supported type for the specific serializer. For `ProtocolBuffers` this is `protobuf::Message`.

Introducing the `M` has other issues I believe, but I couldn't figure out a better way to express "a serializer defines its own message type trait constraint" pattern.

Reply:
I've just made a commit removing serde requirements from most places:
https://github.com/lunatic-solutions/lunatic-rs/pull/64/commits/65acdedeca3248ee6127c0939d01b47f66a4efa5

What do you think of this? Does it solve it how you imagined? I didn't add `M` generic as mentioned, instead I kept `Self::Args` but removed the serde bounds there.

Reply:
If there is no serde requirement, how does the compiler know what types are safe to be used in `Self::Arg`? Not everything is serializable.

Reply:
Hmm, things would be required to be Serialize / Deserialize if used with `Supervisable` with a serializer other than protobuf.
https://github.com/lunatic-solutions/lunatic-rs/blob/65acdedeca3248ee6127c0939d01b47f66a4efa5/src/supervisor.rs#L170-L176